source/game/kart/KartState.cc

```diff
@@ -73,6 +73,14 @@ void KartState::reset() {
 void KartState::calcInput() {
     const auto *raceMgr = System::RaceManager::Instance();
     if (raceMgr->isStageReached(System::RaceManager::Stage::Race)) {
+        if (state()->isCannonStart() || state()->isInCannon()) {
+            // what if a CT author puts a cannon at the starting line?? idk, but this is what ghidra
```

Comment:
This is not capturing the scenario where we are at the starting line (by that, I assume you mean fade-in or countdown). Separately, this doesn't quite match Ghidra - note that there is only one call to `calcHandleStartBoost` in the base game. This should be refactored such that we only process the input state IF we're not in a cannon